### PR TITLE
ARROW-4876: Port MutableBuffer to csharp

### PR DESCRIPTION
**Pull request by pgovind**
Mutable Arrow Buffer implementation

### CODE REVIEW

**Comment by pgovind** [comment]
@eerhardt @chutchinson

File: csharp/src/Apache.Arrow/Ipc/MemoryBufferAllocator.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+using System;
+using FlatBuffers;
+
+namespace Apache.Arrow.Ipc
+{
+    internal sealed class MemoryBufferAllocator : ReadOnlyMemoryBufferAllocator
+    {
+        public MemoryBufferAllocator(Memory<byte> buffer) : base(buffer) { }
+
+        // since this is read-only, the following are not supported
```
Comment (pgovind) [inline csharp/src/Apache.Arrow/Ipc/MemoryBufferAllocator.cs]: Remove comment

**Comment by wesm** [comment]
Did you mean to close this?

**Comment by pgovind** [comment]
Closing this PR. It looks like this is unnecessarry. Clients can change the ReadOnlyMemory to Memory using a call in MemoryMarshal.

**Comment by wesm** [comment]
I see

**Pull request closed without merging**